js/post/fill_textboxes.js

```diff
@@ -1,48 +1,49 @@
 /**
- * Fills textboxes for each POSt for 300 and 400 level categories
- *
- * @param {number} specBound Number of textboxes for specialist
- * @param {number} majBound Number of textboxes for major
- * @param {string} textboxLevel Level of textboxes we are filling
- * @param {string} activeLevel Level of courses we are filling textboxes with
- * @param {number} i The textbox we are at now
- * @param {HTMLElement} specElement The textboxes for Specilist to fill
- * @param {HTMLElement} majElement The textboxes for Major to fill
- * @param {HTMLElement} minElement The textboxes for Minor to fill
- * @return {number} The next textbox that has not been filled 
+ * Fills textboxes in 300 level category,
+ * @param {object} post Object corresponding to the POSt being dealt with
+ * @param {HTMLElement[]} postElement Array of textboxes to fill
+ * @param {string} category Level of course that we are filling textbox with 
 **/
-function fill300and400Textboxes(specBound, majBound, textboxLevel, activeLevel, i, specElement, majElement, minElement) {
+function fill300Textboxes(post, postElement, category) {
     'use strict';
 
-    var index = 'index' + activeLevel;
-
-    for (var m = 0; m < specBound; m++) {
-        if (window[index].spec === window['active' + activeLevel + 's'].length || i === specBound) {
+    for (var m = post.index300 + 1; m < activeCourses.length; m++) {
+        if (post.index300 === activeCourses.length || post.filledTextboxes300 === post.textboxes300) {
```

Comment:
1. post.index300 seems tied to m. This means the first clause is either redundant, or should be folded into the loop.
2. It doesn't look like either `post.filledTextboxes300` or `post.textboxes300` change during the loop. If this is true, this clause shouldn't be in the loop at all.